File: creational/python/builder/Factory.py
from CarBuilder import CarBuilder
from Car import  Car
class Factory:
    ___carBuilder: CarBuilder = None

    def __init__(self, carBuilder: CarBuilder):
        self.___carBuilder = carBuilder

    def getCar(self)->Car:
        return self.___carBuilder.getCar()

    def addAttributes(self):
        self.___carBuilder.buildEngine()
        self.___carBuilder.buildHeadLight()
        self.___carBuilder.buildSunroof()
        self.___carBuilder.buildInterior()

File: creational/java/builder/src/Factory.java
public class Factory {
    private CarBuilder carBuilder;
    public Factory(CarBuilder carBuilder) {
        this.carBuilder = carBuilder;
    }

    public Car getCar(){
        return this.carBuilder.getCar();
    }

    public void addAttributes(){
        this.carBuilder.buildEngine();
        this.carBuilder.buildHeadLight();
        this.carBuilder.buildInterior();
        this.carBuilder.buildSunroof();
    }
}
